api/v1alpha1/envoyproxy_types.go

```diff
@@ -139,6 +163,16 @@ type EnvoyProxyKubernetesProvider struct {
 	EnvoyHpa *KubernetesHorizontalPodAutoscalerSpec `json:"envoyHpa,omitempty"`
 }
 
+// EnvoyProxyDeploymentSpec defines the desired state of the Kubernetes deployment resource.
```

Comment:
suggest not adding this in the first iteration, in case we need to take another approach in the future
the user will still be able to change settings using  the new `patch` field in https://gateway.envoyproxy.io/latest/api/extension_types/#kubernetesdeploymentspec

Reply:
I keep forgetting about that patch field :) updated in cf106fa, modifying params on this container shouldn't be a common case outside of the timeouts anyways